### ISSUE TITLE
[Button] Color specs update

### ISSUE THREAD

**Issue by aweell** [dxc-technology/halstack-style-guide#295]
## Story description

- [x] Update component documentation page with corrected color specs

### React
```
    primaryBackgroundColor: globalTokens.purple,
    primaryBackgroundColorOnDark: "#581B81",
    primaryHoverBackgroundColor: globalTokens.black,
    primaryHoverBackgroundColorOnDark: "#581B81",
    primaryFontColor: globalTokens.white,
    primaryFontColorOnDark: globalTokens.white,
    primaryHoverFontColor: globalTokens.white,
    primaryHoverFontColorOnDark: globalTokens.black,
    primaryActiveBackgroundColor: globalTokens.mediumBlack,
    primaryActiveBackgroundColorOnDark: "#581B81",
    primaryDisabledBackgroundColor: globalTokens.lighterPurple,
    primaryDisabledBackgroundColorOnDark: "#ffffff57",
    primaryDisabledFontColor: globalTokens.white,
    primaryDisabledFontColorOnDark: globalTokens.black,
    primaryBorderColor: globalTokens.black,
    primaryBorderColorOnDark: globalTokens.white,
    secondaryOutlinedColor: globalTokens.purple,
    secondaryOutlinedColorOnDark: "#6A229A",
    secondaryBackgroundColor: globalTokens.transparent,
    secondaryBackgroundColorOnDark: globalTokens.transparent,
    secondaryHoverOutlinedColor: globalTokens.black,
    secondaryHoverOutlinedColorOnDark: "#6A229A",
    secondaryFontColor: globalTokens.black,
    secondaryFontColorOnDark: globalTokens.white,
    secondaryHoverFontColor: globalTokens.black,
    secondaryHoverFontColorOnDark: globalTokens.white,
    secondaryActiveBackgroundColor: globalTokens.mediumGreyBlack,
    secondaryActiveBackgroundColorOnDark: "#FFFFFF29",
    secondaryHoverBackgroundColor: globalTokens.darkWhite,
    secondaryHoverBackgroundColorOnDark: "#FFFFFF14",
    secondaryDisabledOutlinedColor: globalTokens.lighterPurple,
    secondaryDisabledOutlinedColorOnDark: "#575757",
    secondaryDisabledFontColor: globalTokens.lighterBlack,
    secondaryDisabledFontColorOnDark: "#575757",
    textBackgroundColor: globalTokens.transparent,
    textBackgroundColorOnDark: globalTokens.transparent,
    textHoverBackgroundColor: globalTokens.black,
    textHoverBackgroundColorOnDark: "#6A229A80",
    textFontColor: globalTokens.purple,
    textFontColorOnDark: globalTokens.white,
    textHoverFontColor: globalTokens.white,
    textHoverFontColorOnDark: globalTokens.white,
    textActiveBackgroundColor: globalTokens.mediumBlack,
    textActiveBackgroundColorOnDark: "#581B81",
    textDisabledFontColor: globalTokens.lighterPurple,
    textDisabledBackgroundColor: globalTokens.transparent,
    textDisabledBackgroundColorOnDark: globalTokens.transparent,
    textDisabledFontColorOnDark: "#FFFFFF80",
    textBorderColor: globalTokens.black,
    textBorderColorOnDark: globalTokens.white,
    focusColor: globalTokens.blue,
```
   
### Angular (Pending Update)

```

```

##### Base

| Component token                                | Element                                        | Token                       | Value           |
| ------------------------------------ | ------------------------------- | ------------------- | --------------- |
| `primaryBackgroundColor`                | Button container                         | `Hal-Purple-S-38` | `#5f249f`          |
| ` primaryFontColor`                            | Label                                            | `Hal-White`             | `#ffffff`             |

##### Interactive

| Component token              | Element                     | Token             | Value           |
| ------------------------------------ | --------------------------- | ----------------- | --------------- |
| `primaryHoverBackgroundColor`      | Container fill:hover              | `Hal-Purple-D-30`   | `#4b1c7d`        |
| `primaryActiveBackgroundColor`      | Container fill:active             | `Hal-Purple-D-20`   | `#321353`        |
| `primaryDisabledBackgroundColor`  | Container fill:disabled         | `Hal-Grey-L-95`      | `#f2f2f2`          |
| ` primaryDisabledFontColor`             | Label:disabled                      | `Hal-Grey-L-60`      | `#999999`       |

##### Base

| Component token                                | Element                                        | Token                       | Value           |
| ------------------------------------ | ------------------------------- | ------------------- | --------------- |
| `secondaryBackgroundColor`           | Container fill                                | `transparent`           | `transparent`   |
| `secondaryFontColor`                        | Label                                            | `Hal-Purple-S-38` | `#5f249f`          |
| `secondaryBorderColor`                    | Container border                        | `Hal-Purple-S-38` | `#5f249f`          |

##### Interactive

| Component token              | Element                     | Token             | Value           |
| --------------------------------------- | --------------------------- | ----------------- | --------------- |
| `secondaryHoverBackgroundColor`      | Container fill:hover              | `Hal-Purple-L-95`   | `#f2eafa`        |
| `secondaryActiveBackgroundColor`      | Container fill:active             | `Hal-Purple-L-90`   | `#e5d5f6`        |
| `secondaryDisabledFontColor`              | Label:disabled                            | `Hal-Grey-L-60`      | `#999999`       |
| `secondaryDisabledBorderColor`          | Container border:disabled         | `Hal-Grey-L-60`      | `#999999`          |

##### Base

| Component token                                | Element                                        | Token                       | Value           |
| ------------------------------------ | ------------------------------- | ------------------- | --------------- |
| `textBackgroundColor`                      | Container fill                                | `transparent`           | `transparent`   |
| `textFontColor`                                  | Label                                            | `Hal-Purple-S-38` | `#5f249f`          |

##### Interactive

| Component token              | Element                     | Token             | Value           |
| --------------------------------------- | --------------------------- | ----------------- | --------------- |
| `textHoverBackgroundColor`      | Container fill:hover              | `Hal-Purple-L-95`   | `#f2eafa`        |
| `textActiveBackgroundColor`      | Container fill:active             | `Hal-Purple-L-90`   | `#e5d5f6`        |
| `textDisabledBackgroundColor`  | Container fill:disabled         | `Hal-Grey-L-95`      | `#f2f2f2`          |
| `textDisabledFontColor`              | Label:disabled                            | `Hal-Grey-L-60`      | `#999999`       |

**Comment by GomezIvann** [comment]
Changes made on the Accordion advanced tokens:

- [ ] `primaryBorderColor` has a new value: `#000000`.
- [ ] `primaryBorderColorOnDark` has a new value: `#FFFFFF`.
- [ ] `textBorderColor` has a new value: `#000000`.
- [ ] `textBorderColorOnDark` has a new value: `#FFFFFF`.

**Comment by GauravMaurya0** [comment]
## React
  
primaryBackgroundColor: globalTokens.purple,
primaryBackgroundColorOnDark: "#581B81",
primaryHoverBackgroundColor: globalTokens.black,
primaryHoverBackgroundColorOnDark: "#581B81",
primaryFontColor: "#FFFFFF",
primaryFontColorOnDark: "#FFFFFF",
primaryHoverFontColor: globalTokens.white,
primaryHoverFontColorOnDark: "#000000",
primaryActiveBackgroundColor: globalTokens.mediumBlack,
primaryActiveBackgroundColorOnDark: "#581B81",
primaryDisabledBackgroundColor: globalTokens.lighterPurple,
primaryDisabledBackgroundColorOnDark: "#ffffff57",
primaryDisabledFontColor: globalTokens.white,
primaryDisabledFontColorOnDark: "#000000",

primaryBorderColor: "",
primaryBorderColorOnDark: "",
secondaryOutlinedColor: globalTokens.purple,
secondaryOutlinedColorOnDark: "#6A229A",
secondaryBackgroundColor: globalTokens.transparent,
secondaryBackgroundColorOnDark: "transparent",
secondaryHoverOutlinedColor: globalTokens.black,
secondaryHoverOutlinedColorOnDark: "#6A229A",
secondaryFontColor: globalTokens.black,
secondaryFontColorOnDark: "#FFFFFF",
secondaryHoverFontColor: globalTokens.black,
secondaryHoverFontColorOnDark: "#FFFFFF",
secondaryActiveBackgroundColor: globalTokens.mediumGreyBlack,
secondaryActiveBackgroundColorOnDark: "#FFFFFF29",
secondaryHoverBackgroundColor: globalTokens.darkWhite,
secondaryHoverBackgroundColorOnDark: "#FFFFFF14",
secondaryDisabledOutlinedColor: globalTokens.lighterPurple,
secondaryDisabledOutlinedColorOnDark: "#575757",
secondaryDisabledFontColor: globalTokens.lighterBlack,
secondaryDisabledFontColorOnDark: "#575757",
textBackgroundColor: globalTokens.transparent,
textBackgroundColorOnDark: "transparent",
textHoverBackgroundColor: globalTokens.black,
textHoverBackgroundColorOnDark: "#6A229A80",
textFontColor: globalTokens.purple,
textFontColorOnDark: "#FFFFFF",
textHoverFontColor: globalTokens.white,
textHoverFontColorOnDark: "#FFFFFF",
 textActiveBackgroundColor: globalTokens.mediumBlack,
 textActiveBackgroundColorOnDark: "#581B81",
 textDisabledFontColor: globalTokens.lighterPurple,
 textDisabledBackgroundColor: "transparent",
 textDisabledBackgroundColorOnDark: "transparent",
 textDisabledFontColorOnDark: "#FFFFFF80",
 textBorderColor: "",
 textBorderColorOnDark: "",
 focusColor: globalTokens.blue,
  
## Angular
  
"--button-primaryBackgroundColor": globalTokens.purple,
"--button-primaryHoverBackgroundColor": globalTokens.black,
"--button-primaryFontColor": globalTokens.white,
"--button-primaryHoverFontColor": globalTokens.white,
"--button-disabledPrimaryBackgroundColor": globalTokens.lighterPurple,
"--button-primaryActiveBackgroundColor": globalTokens.mediumBlack,
"--button-disabledPrimaryFontColor": globalTokens.white,
"--button-secondaryFontColor": globalTokens.black,
"--button-secondaryHoverFontColor": globalTokens.black,
"--button-secondaryBackgroundColor": globalTokens.transparent,
"--button-secondaryOutlinedColor": globalTokens.purple,
"--button-hoverOutlinedColor": globalTokens.black,
"--button-secondaryActiveBackgroundColor": globalTokens.mediumGreyBlack,
"--button-secondaryHoverBackgroundColor": globalTokens.darkWhite,
"--button-disabledSecondaryOutlinedColor": globalTokens.lighterPurple,
"--button-disabledSecondaryFontColor": globalTokens.lighterBlack,
"--button-textHoverBackgroundColor": globalTokens.black,
"--button-textActiveBackgroundColor": globalTokens.mediumBlack,
"--button-textFontColor": globalTokens.purple,
"--button-textHoverFontColor": globalTokens.white,
"--button-textBackgroundColor": globalTokens.transparent,
"--button-disabledTextFontColor": globalTokens.lighterPurple,



## Button color specs table

### Primary
#### On-light

| Component token                         | Element       | Core token | Value |
| -----------------------------   | ----------- | -------- | --------- |
| `primaryBackgroundColor` | Fill | `Hal-Purple-S-38` | `#5f249f` |
| `primaryHoverBackgroundColor` | Fill:hover | `Hal-Black` | `#000000` |
| `primaryFontColor` | Label | `Hal-White` | `#ffffff` |
| `primaryHoverFontColor` | Label:hover | `Hal-White` | `#ffffff` |
| `primaryActiveBackgroundColor` | Fill:active | `Hal-Grey-S-40` | `#666666` |
| `disabledPrimaryBackgroundColor` | Fill:disabled | `Hal-Purple-L-90` | `#e5d5f6` |
| `disabledPrimaryFontColor` | Label:disabled | `Hal-White` | `#ffffff` |
| `primaryBorderColor` | Border | - | - |

#### On-dark

| Component token                         | Element       | Core token | Value |
| -----------------------------   | ----------- | -------- | --------- |
| `primaryBackgroundColorOnDark` | Fill | `Hal-Purple-S-38` | `#5f249f` |
| `primaryHoverBackgroundColorOnDark` | Fill:hover | `Hal-Purple-S-38` | `#5f249f` |
| `primaryFontColorOnDark` | Label | `Hal-White` | `#ffffff` |
| `primaryHoverFontColorOnDark` | Label:hover | `Hal-Black` | `#000000` |
| `primaryActiveBackgroundColorOnDark` | Fill:active | `Hal-Purple-S-38` | `#5f249f` |
| `disabledPrimaryBackgroundColorOn Dark` | Fill:disabled | - | - |
| `disabledPrimaryFontColorOnDark` | Label:disabled | `Hal-Black` | `#000000` |
| `primaryBorderColorOnDark` | Border | - | - |


### Secondary
#### On-light

| Component token                         | Element       | Core token | Value |
| -----------------------------   | ----------- | -------- | --------- |
| `secondaryFontColor` | Label | `Hal-Black` | `#000000` |
| `secondaryHoverFontColor` | Label:hover | `Hal-Black` | `#000000` |
| `secondaryActiveBackgroundColor` | Fill:active | `Hal-Grey-S-40` | `#666666` |
| `secondaryHoverBackgroundColor` | Fill:hover | `Hal-White` | `#ffffff` |
| `secondaryOutlinedColor` | Border:hover | `Hal-Purple-S-38` | `#5f249f` |
| `secondaryBackgroundColor` | Fill | - | `transparent` |
| `secondaryHoverOutlinedColor` | Fill:hover | `Hal-Black` | `#000000` |
| `disabledSecondaryOutlinedColor` | Fill:disabled | `Hal-Purple-L-90` | `#e5d5f6` |
| `disabledSecondaryFontColor` | Label:disabled | `Hal-Grey-L-60` | `#999999` |

#### On-dark

| Component token                         | Element       | Core token | Value |
| -----------------------------   | ----------- | -------- | --------- |
| `secondaryFontColorOnDark` | Label | `Hal-White` | `#ffffff` |
| `secondaryHoverFontColorOnDark` | Label:hover | `Hal-White` | `#ffffff` |
| `secondaryActiveBackgroundColorOnDark` | Fill:active | - | - |
| `secondaryHoverBackgroundColorOnDark` | Fill:hover | - | - |
| `secondaryOutlinedColorOnDark` | Border:hover | - | - |
| `secondaryBackgroundColorOnDark` | Fill | - | `transparent` |
| `secondaryHoverOutlinedColorOnDark` | Fill:hover | `Hal-Purple-S-38` | `#5f249f` |
| `secondaryDisabledOutlinedColorOnDark` | Fill:disabled | `Hal-Grey-S-40` | `#666666` |
| `disabledSecondaryFontColorOnDark` | Label:disabled | `Hal-Grey-S-40` | `#666666` |


### Text
#### On-light

| Component token                         | Element       | Core token | Value |
| -----------------------------   | ----------- | -------- | --------- |
| `textBackgroundColor` | Fill | - | `transparent` |
| `textHoverBackgroundColor` | Fill:hover | `Hal-Black` | `#000000` |
| `textFontColor` | Label | `Hal-Purple-S-38` | `#5f249f` |
| `textHoverFontColor` | Label:hover | `Hal-White` | `#ffffff` |
| `textActiveBackgroundColor` | Label:active | `Hal-Grey-S-40` | `#666666` |
| `disabledTextFontColor` | Label:disabled | `Hal-Purple-L-90` | `#e5d5f6` |
| `disabledtextBackgroundColor` | LabelFill:disabled | - | `transparent` |
| `textBorderColor` | Label:border | - | - |
| `focusColor` | Outline:focus | `Hal-Blue-L-50` | `#0095ff ` |

#### On-dark

| Component token                         | Element       | Core token | Value |
| -----------------------------   | ----------- | -------- | --------- |
| `textBackgroundColorOnDark` | Fill | - | `transparent` |
| `textHoverBackgroundColorOnDark` | Fill:hover | - | - |
| `textFontColorOnDark` | Label | `Hal-White` | `#ffffff` |
| `textHoverFontColorOnDark` | Label:hover | `Hal-White` | `#ffffff` |
| `textActiveBackgroundColorOnDark` | Label:active | `Hal-Purple-S-38` | `#5f249f` |
| `disabledtextBackgroundColorOnDark` | LabelFill:disabled | - | `transparent` |
| `disabledtextFontColorOnDark` | Label:disabled | - | - |
| `textBorderColorOnDark` | Label:border | - | - |

**Comment by aweell** [comment]
> * [ ]  `primaryBorderColor` has a new value: `#000000`.
> - [ ] `textBorderColorOnDark` has a new value: `#FFFFFF`.


When this tokens apply?

There's no border in any of the enabled states for this button variants

**Comment by GomezIvann** [comment]
> > * [ ]   `primaryBorderColor` has a new value: `#000000`.
> > 
> > 
> > * [ ]  `textBorderColorOnDark` has a new value: `#FFFFFF`.
> 
> When this tokens apply?
> 
> There's no border in any of the enabled states for this button variants

By default, our buttons don't have any kind of border. But, by using our tokens you can add them one. Here it's an example with `primaryBorderXXX` tokens, which are being use to add a border to the Primary Button:

<img width="1268" alt="Sin título" src="https://user-images.githubusercontent.com/44321109/125403963-62701e00-e3b6-11eb-8689-44f5d97dbae6.png">

The same applies to the `textBorderXXX` tokens, which do the exact same thing but with the Text Button mode:

![image](https://user-images.githubusercontent.com/44321109/125403619-02797780-e3b6-11eb-90ad-4f14c3325a8a.png)

And also , the `secondaryBorderXXX` tokens for the Secondary Button.

**Comment by aweell** [comment]
Closed via #433

**Comment by aweell** [comment]
## [3.6.0] Update

* Removed tokens for label:active

**Comment by aweell** [comment]
Closed via #448

**Comment by aweell** [comment]
Fixes needed on interactive tokens (#448 wasn't applied correctly):

- [ ] Remove `:hover` and `:active` font color tokens
- [ ] Update `secondaryDisabledBackgroundColor` value to transparent

**Comment by aweell** [comment]
Closed via #494
